tests/upload_files.rs

```diff
@@ -33,7 +43,12 @@ fn uploading_files_works(#[with(&["-u"])] server: TestServer) -> Result<(), Erro
         .mime_str("text/plain")?;
     let form = form.part("file_to_upload", part);
 
-    let client = Client::new();
+    let mut headers = HeaderMap::new();
+    headers.insert("X-File-Hash", sha.parse()?);
+    headers.insert("X-File-Hash-Function", sha_func.parse()?);
```

Comment:
I think it would be more proper to make this test behave in the same way a client leaving out the headers would do it. I suppose it would therefore be more realistic to make the parameters into `Option<String>`s?

Frankly I'm not sure how this currently passes since we're setting the headers but we just leave the values as an empty string.

Reply:
It passes because the endpoint isn't programmed to fail if incorrect values are passed in, instead, it will continue on silently. It might be worthwhile to update it so that if the headers are in the HTTP request, then it's enforced they have valid values.